aiohttp/helpers.py

```diff
@@ -47,6 +47,10 @@ def __new__(cls, login, password='', encoding='latin1'):
         if password is None:
             raise ValueError('None is not allowed as password value')
 
+        if ':' in login:
+            raise ValueError(
+                'A ":" is not allowed in login (RFC 1945#section-11.1)')
```

Comment:
Sorry so being late for the party, but this is not very correct RFC reference since: 
1) it about HTTP 1/0 
2) there it disallows non-latin usernames, while we don't.
Here is a better one: https://tools.ietf.org/html/rfc2617#section-2